portfolio/src/main/webapp/script.js

```diff
@@ -22,6 +21,47 @@ var counter = 1;
 var intervalLength = setInterval(change, 2000);
 document.getElementById('commentBtn').addEventListener('click', getComments);
 
+var script = document.createElement('script');
```

Comment:
Instead of putting this in via JS, can you just put it in statically in the HTML?

Reply:
If it doesn't work, you might be hitting a subtle race condition that the tutorials never discuss.

Background: HTML parsers (e.g. browser) process HTML in source code order. When they hit a script tag, they stop parsing until the script downloads and executes, by default. (If defer or async is present, they may continue accordingly, see the docs.)

Race condition: This gmaps script tag in the head is downloaded but since it's async, the browser keeps going until it hits the end of body and executes this script we're in which defines initMap. The race condition is which script executes first - this one needs to execute first to define initMap for the gmaps script not to error. The 2nd time onward, it'll probably always error since the gmaps script will be cached and will execute near immediately. So consider ordering your script tags accordingly to avoid this.